src/MMALSharp.Processing/Handlers/ExternalProcessCaptureHandler.cs

```diff
@@ -0,0 +1,330 @@
+﻿// <copyright file="ExternalProcessCaptureHandler.cs" company="Techyian">
+// Copyright (c) Ian Auty. All rights reserved.
+// Licensed under the MIT License. Please see LICENSE.txt for License info.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using MMALSharp.Common;
+using MMALSharp.Common.Utility;
+using Mono.Unix.Native;
+
+namespace MMALSharp.Handlers
+{
+    /// <summary>
+    /// This handler is a general-purpose manager for external processes like ffmpeg and VLC.
+    /// It uses Channel-based async buffering of console output, and supports options to properly
+    /// terminate the child process through one or more signals (SIGINT, SIGQUIT, etc.).
+    /// </summary>
+    public class ExternalProcessCaptureHandler : IVideoCaptureHandler
+    {
+        private readonly ExternalProcessCaptureHandlerOptions _options;
+        private readonly Process _process;
+        private readonly Channel<string> _stdoutBuffer;
+
+        /// <summary>
+        /// The total size of data that has been processed by this capture handler.
+        /// </summary>
+        protected int Processed { get; set; }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="ExternalProcessCaptureHandler"/> with the specified options.
+        /// </summary>
+        public ExternalProcessCaptureHandler(ExternalProcessCaptureHandlerOptions options)
+        {
+            if(MMALLog.Logger != null)
```

Comment:
You don't need to include the null check as the `MMALLog.Logger` class handles this on your behalf so you can safely remove.